juju/framework.py

```diff
@@ -537,8 +548,15 @@ def __init__(self, parent, attr_name):
         except NoSnapshotError:
             data = StoredStateData(parent, attr_name)
 
+        # __dict__ is used to avoid infinite recursion.
         self.__dict__["_data"] = data
         self.__dict__["_attr_name"] = attr_name
+        self.__dict__["handle"] = handle
```

Comment:
One more detail: I don't think this should be here, and the event handler shouldn't be here either. Both should be in the StoredStateData, which is already a subclass of Object, and already has a handle of its own.

Reply:
Good point, thank you.